core/shared/src/main/scala/org/http4s/metrics/MetricsOps.scala

```diff
@@ -18,57 +18,189 @@ package org.http4s.metrics
+    * @param uri the URI of the request
+    * @param protocol the protocol of the request
+    * @param address the address of the local server
+    * @param status the status of the response
+    * @param terminationType the termination type
+    * @param elapsed the processing time
     * @param classifier the classifier to apply
     */
   def recordTotalTime(
       method: Method,
-      status: Status,
-      elapsed: Long,
+      uri: Uri,
+      protocol: NetworkProtocol,
+      address: Option[SocketAddress[IpAddress]],
+      status: Option[Status],
+      terminationType: Option[TerminationType],
+      elapsed: FiniteDuration,
       classifier: Option[String],
   ): F[Unit]
 
-  /** Record abnormal terminations, like errors, timeouts or just other abnormal terminations.
+  /** Records the size of the request body
+    *
+    * {{{
+    * | Name                      | Example                         | Requirement level                                            |
+    * |---------------------------|---------------------------------|--------------------------------------------------------------|
+    * | http.request.method       | `GET`, `POST`                   | Required                                                     |
+    * | url.scheme                |  `http`, `https`                | Required                                                     |
+    * | error.type                | `java.net.UnknownHostException` | Required If request has ended with an error                  |
+    * | http.response.status_code | `200`, `404`                    | Conditionally Required If and only if one was received/sent. |
+    * | network.protocol.name     |  `http`, `spdy`                 | Conditionally Required                                       |
+    * | network.protocol.version  | `1.0`, `1.1`, `2`, `3`          | Recommended                                                  |
+    * | server.address            | `example.com`, `10.1.2.80`      | Opt-In                                                       |
+    * | server.port               | `80`, `8080`                    | Opt-In                                                       |
+    * }}}
     *
-    * @param elapsed the time to record
-    * @param terminationType the type of termination
+    * @param method the http method of the request
+    * @param uri the URI of the request
+    * @param protocol the protocol of the request
+    * @param address the address of the local server
+    * @param status the status of the response
+    * @param terminationType the termination type
+    * @param contentLength the size of the request body
     * @param classifier the classifier to apply
     */
-  def recordAbnormalTermination(
-      elapsed: Long,
-      terminationType: TerminationType,
+  def recordRequestBodySize(
+      method: Method,
+      uri: Uri,
+      protocol: NetworkProtocol,
+      address: Option[SocketAddress[IpAddress]],
```

Comment:
Almost all methods in the `MetricsOps` algebra require `method`, `uri`, `protocol`, and `address`.
It makes signatures clunky and hard to read. 

What if we:
a) Pass `Request[F]` instead - simple and effective
b) Introduce `RequestInfo` (or something similar) - we will deal with additional allocations